src/functions/utils.php

```diff
@@ -645,19 +645,41 @@ function tribe_register_rest_route( $namespace, $route, $args = array(), $overri
  *
  * @since TBD
  *
- * @param object $class The plugin object.
+ * @param string|object $class The plugin's main class name or its instance.
  *
- * @return string|boolean The SemVer version string, or false if no info found.
+ * @return string|boolean The SemVer version string or false if no info found.
  */
 function tribe_get_install_version( $class ) {
 	// Try for the version history first.
-	if ( ! empty( $class->version_history_slug ) ) {
-		return Tribe__Settings_Manager::get_option( $class->version_history_slug );
+	if ( is_string( $class ) ) {
+		if ( method_exists( $class, 'instance' ) ) {
+			$instance = $class::instance();
+		} elseif ( method_exists( $class, 'get_instance' ) ) {
+			$instance = $class::get_instance();
+		} else {
+			$instance = '';
+		}
+	} else {
+		$instance = $class;
+	}
+
+	if ( ! empty( $instance->version_history_slug ) ) {
+		$history = (array) Tribe__Settings_Manager::get_option( $instance->version_history_slug );
+
+		// '0' may be logged as a version number, which isn't useful, so we remove it
+		$history = array_filter( $history );
+
+		// Sort the array so smallest version number is first (likely how the array is stored anyway)
+		usort( $history, 'version_compare' );
```

Comment:
Why sort? Why not just use the last one?

Reply:
You mean the first one that's not a zero?

We're looking for the oldest-registered version number.

Reply:
The zero was removed via array_filter() already though, right?

So wouldn't you just check `$history[0]`?

Reply:
```
// array_filter removes the '0' item from the beginning if it exists
$history = array_filter( $history );

if ( empty( $history ) ) {
	return false;
}

// Gets the first item in the array - the last added, therefore the oldest.
$intall_version = array_shift($history);

if (  ! empty( $intall_version ) ) {
	return $intall_version;
}

return false;
```

Reply:
You could also just:
```
while ( ! empty( $history ) && empty( $history[0] ) ) {
    unset( $history[0] );
}

// just in case ;)
if ( empty( $history[0] ) ) {
	return false;
}

// We found it, return it
if (  ! empty( $history[0] ) ) {
	return $history[0];
}

// fallback
return false;
```

Reply:
I mean you could just take out the 'just in case' conditional at that point :)

Reply:
True - edited 😛

Reply:
@sc0ttkclark, is it possible someone installs version 2, finds a known issue, downgrades to version 1, then first entry is `2` but oldest version (from sorting) is actually `1`?

Reply:
That depends on our intention here, if we want the oldest/first installed version on a new installation, that would be v2. It just depends on what we want intent to be on this function.

Reply:
i guess that's a question for @Camwyn, as he was the one who originally wrote these new functions... the function name and docblock should be very clear either way you/team decide.

Reply:
That depends on how the history is used - do we record changes back and forth? Do we _use_ that info?

Maybe the best answer here is to modify where we store it to sort and only store unique values?

Reply:
For my purpose, I want to know the earliest version ever installed, although the first version ever installed is likely the same for nearly all installs. If someone's initial install is v4.11.0, but for some reason decided to use 4.10.99 instead for a period of time, I _do_ want to send them a notice because they were using a version prior to 4.11.0

If this is a different use case for why these functions were created, I can do it my own way - just need to know these comparison functions' final decision so I can decide whether or not to use them.

Reply:
@cliffordp I'm curious, what's the use case?

Reply:
The attached Central ticket and ET+ pull request

Reply:
At this point, for sanity's sake, I think the comparison should do something like:
```
$history = array_filter( $history );
$history = array_unique($history );
usort( $history, 'version_compare' );
```
Before doing comparisons.

Probably in a separate small function? So we can just call `version_history_clean($history);` and then `array_shift($history)` and do our comparison?

Reply:
As per latest slack discussion: https://tribe.slack.com/archives/CHXD8BSH4/p1567698140042700

- [ ] Clarify that this function will return the *first* installed version, not the *earliest* version number and rename function to `tribe_get_first_installed_version`
- [ ] Make the change that @Camwyn suggested in https://github.com/moderntribe/tribe-common/pull/1139#discussion_r320798585 to this function
- [ ] Adjust the code formatting in this function, the tabs/spaces look off
- [ ] Replace any usage of `tribe_get_install_version` to use `tribe_get_first_installed_version` instead
- [ ] Create a new function that will return the earliest version number using the sorting logic `tribe_get_earliest_installed_version`